app/com/arpnetworking/kairos/client/models/MetricsQuery.java

```diff
@@ -0,0 +1,547 @@
+        return _endTime.getMillis();
+    }
+
+    @JsonProperty("metrics")
+    public List<Metric> getMetrics() {
+        return _metrics;
+    }
+
+    private MetricsQuery(final Builder builder) {
+        _startTime = builder._startTime;
+        _endTime = builder._endTime;
+        _metrics = ImmutableList.copyOf(builder._metrics);
+    }
+
+    private final DateTime _startTime;
+    private final DateTime _endTime;
+    private final List<Metric> _metrics;
+
+    /**
+     * Implementation of the builder pattern for MetricsQuery.
+     */
+    public static final class Builder extends OvalBuilder<MetricsQuery> {
+        /**
+         * Public constructor.
+         */
+        public Builder() {
+            super(MetricsQuery::new);
+        }
+
+        /**
+         * Sets the start time of the query. Required. Cannot be null.
+         *
+         * @param value the start time
+         * @return this {@link Builder}
+         */
+        public Builder setStartTime(final DateTime value) {
+            _startTime = value;
+            return this;
+        }
+
+        /**
+         * Sets the end time of the query. Optional. Cannot be null.  Default is now.
+         *
+         * @param value the end time
+         * @return this {@link Builder}
+         */
+        public Builder setEndTime(final DateTime value) {
+            _endTime = value;
+            return this;
+        }
+
+        /**
+         * Sets list of metrics. Required. Cannot be null or empty.
+         *
+         * @param value the metrics
+         * @return this {@link Builder}
+         */
+        public Builder setMetrics(final List<Metric> value) {
+            _metrics.clear();
+            _metrics.addAll(value);
+            return this;
+        }
+
+        /**
+         * Adds a metric to the list of metrics. Cannot be null.
+         *
+         * @param value a metric
+         * @return this {@link Builder}
+         */
+        public Builder addMetric(final Metric value) {
+            _metrics.add(value);
+            return this;
+        }
+
+        /**
+         * Creates a new builder based on this one.
+         *
+         * @return a new builder with the same
+         */
+        public Builder copy() {
+            return new Builder().setStartTime(_startTime)
+                    .setEndTime(_endTime)
+                    .setMetrics(_metrics.stream()
+                            .map(Metric.Builder::from)
+                            .map(Metric.Builder::build)
+                            .collect(Collectors.toList()));
+        }
+
+        public DateTime getStartTime() {
+            return _startTime;
+        }
+
+        public DateTime getEndTime() {
+            return _endTime;
+        }
+
+        public List<Metric> getMetrics() {
+            return _metrics;
+        }
+
+        @NotNull
+        private DateTime _startTime;
+        @NotNull
+        private DateTime _endTime = DateTime.now();
```

Comment:
Not nuts about this default.

Reply:
It's actually what Kairos does, so I don't feel too bad about it.

Reply:
It's one thing for it to do it on the server side and something else for us to do it in our model.

Reply:
Ok, then I'll allow null and exclude it if it is.